visualization-design.typ

```diff
@@ -4,32 +4,26 @@ The visualization serves a dual purpose: To communicate the functionality of the
 
 == Hosting
 
-Option 1: Self-Hosted
-- Pros: No added cost, no compromises required   
-- Cons: Not always online, longer startup time
-
-Option 2: Buy a domain
-- Pros: Always online, easy access
-- Cons: Cost (that's probably enough of a con)
-
-Option 3: Vercel
-- Pros: Easy, free
-- Cons: .vercel in the hostname, ???
-
-Option 4: Another free hosting service
-- Pros: No added cost
-- Cons: Research required, design compromises may be needed, may be hard to setup
+Due to time constraints, the visualization will be self-hosted on the user's machine. If needed, a subdomain of OpenTogetherTube can be established at a later time.
 
 == Frameworks/Libraries
 
-This is just a shortlist, looking at more tomorrow:
-
-- *Cytoscape.js* - I wanna look into this one a little more but just look at this: https://js.cytoscape.org/demos/colajs-graph/
-- D3 - Seems ok, esp force-directed graph. Like how popular/flexible it is, don't like how messy it seems
-- React-vis - Seems promising, only works with react though
-- Plotnine - probably not, just wanted to look at Python libraries
-
-There's always the possibility of writing a graphing library, but that would be a full project in-of-itself
+The visualization must be ready before the innovation exposition on April 26 and is not the main focus of the project. With this in mind, Vue.js and D3.js have been selected for development due to their popularity, ease of use, and low-learning curve.
```

Comment:
I just thought of something: I already use grafana to plot graphs and stuff of metrics. If we could make a grafana plugin to show our custom graphs, that would save us a ton of time.

Reply:
Here's the quick start docs for grafana plugins https://grafana.com/developers/plugin-tools/

Reply:
Just looked through and skimmed around the doc website: That could work, not exactly sure if I'm understanding exactly correct but I saw something about a datasource streaming example plugin made to update visualizations. Maybe that would be a good starting point? Then make that fetched data work with d3, at least for the views with vertices and edges?

Reply:
Oh also, should I setup a new repo for the visualization?

Reply:
Lets keep it in the OTT repo.

Reply:
Will respond with availability privately